rmf_site_editor/src/site/level.rs

```diff
@@ -66,3 +66,33 @@ pub fn assign_orphan_elements_to_level<T: Component>(
         commands.entity(current_level).add_child(orphan);
     }
 }
+
+pub fn set_camera_transform_on_level_change(
```

Comment:
Ah yes good point, this puzzling decision deserves some background.

The main reason is compatibility with the rest of the stack. Among other things, this PR pretty much translates the whole site by the transform of the reference level https://github.com/open-rmf/rmf_site/commit/a16a41fe0896bf96983165911cde712a94d2f851.
This was to make sure that maps exported with this pipeline keep the same coordinates as maps exported with our legacy pipeline, which enforces a reference level with an identity transform.
If we don't do this, any place that hardcodes any sort of coordinates will suddenly not work, among others:

* Hardcoded [paths for cleaning robots](https://github.com/open-rmf/rmf_demos/blob/5ab6a3ddd8681da188610a5784bacf1b0eb3cf5f/rmf_demos/config/hotel/cleanerBotA_config.yaml#L52-L72) in demos.
* Hardcoded Open-RMF coordinates for `nudged` transform estimate in any Python fleet adapter based on the [demo](https://github.com/open-rmf/rmf_demos/blob/5ab6a3ddd8681da188610a5784bacf1b0eb3cf5f/rmf_demos_fleet_adapter/config.yaml#L46-L56) (but reasonably all open source adapters since they are based on this).

Now since the whole map is translated by a significant amount to make sure the reference level is unchanged, startup will look a bit funny, here it is without the system:

![image](https://github.com/open-rmf/rmf_site/assets/9111558/b30c4c29-a32b-425d-bcae-d3ad90fd12f4)

Hence we would need a system that translates the camera to make sure it is in a sensible position.
I tried to address the issue in https://github.com/open-rmf/rmf_site/pull/215/commits/6426ece2a8f3f9fd1718b1d66d9b5f6bbe9ef3e3 where I changed the system run condition to be a change detection on `CurrentWorkspace`, rather than `CurrentLevel`. When `CurrentWorkspace` is changed, I set it as the default pose of the current level.

This should fix the behavior you mentioned while still placing the camera in sensible places when switching between different sites. Next I'll be working on your other piece of feedback to remove camera poses being in a `HashMap` and making them children of `Level` or `Site`.

Reply:
Awesome, thanks for the explanation, now I understand why it was done. And an additional thanks for the fix; I think this is a good solution.